### DISCUSSION TITLE
ODataActionParameters is null when calling an bound action.

### DISCUSSION THREAD

**Discussion by sharathgt** [OData/AspNetCoreOData#670] [General]
```
var fun = builder.EntityType<AFile>().Collection.Action("AStatus");
fun.Parameter<int>("month");
fun.Parameter<int>("year");
fun.Parameter<int>("key");
```

```
[HttpPost]
public async Task<ActionResult> AStatus(ODataActionParameters parameters)
{
            int month = (int)parameters["month"]; 
            ... 
            ...
}
```

The action get hits with no parameter values 

Tried in Postman with below Json Body 
...api/v2/AFiles/Default.AStatus
{  
  "month" : 2,
  "year" : 2022,
  "key" : 5
}
and also with collection parameter.

**Comment by julealgon** [comment]
Is your route being detected as an OData route? Check the `$odata` endpoint and report back.

**Comment by julealgon** [comment]
Also, does your method cause any side effects? It seems like it is a search, which wouldn't cause side effects, just return the results.

In that case, I recommend using a bound function, instead of a bound action. Note that functions shouldn't cause side effects, and are accessed with GET and parameters in the URL.

**Comment by sharathgt** [comment]
@julealgon , I changed from a normal attribute directed routes to make them OData routes by adding bound actions to models. Yes, it is OData route now. But I saw it always null. I googled many solutions but nothing seems to close to my issue.

For now I reverted my code and working. I shall take this up in my cooling period.

**Reply by julealgon** [reply]
If you are able to provide a small repro project that would be nice @sharathgt . You are very probably missing something very small here.

**Comment by sharathgt** [comment]
I had kept it for weekend. And after trying few things out... making it precise by removing all code inside, single parameters in action, I could pass the value, which worked fine. Came to know that the difference in parameter definitions(int while binding) and long(while accepting) was creating the problem. I could then add other things which I removed earlier and now it receives parameters as expected. 

As you told it might have been a small thing, which when rewritten had gone without my notice... but the param type is one which I noticed. Or I don't know if anything else did the trick because I tried out too many things.

Again, Thanks for your time and support.

**Reply by julealgon** [reply]
Glad you managed to make it work on your end!

Remember to take a look at the difference between "actions" and "functions" later as I mentioned. If you are writing a query that has no side effects, you might want to rely on functions instead of actions.

I know you were just trying to make it work though, so this is for future cases where you have to consider creating a custom action/function.
